### ISSUE TITLE
<code> blocks are converted incorrectly

### ISSUE THREAD

**Issue by scssyworks** [showdownjs/showdown#420]
Example:

Markdown:
&lt;code&gt;Hello &amp;lt;World&amp;gt;&lt;/code&gt;

HTML conversion:
&lt;p&gt;&lt;code&gt;Hello &amp;amp;lt;World&amp;amp;gt;&lt;/code&gt;&lt;/p&gt;

&amp; gets converted to &amp;amp; <-- Not Expected

If we remove the code block, same gets converted to this:

&lt;p&gt;Hello &amp;lt;World&amp;gt;&lt;/p&gt;

As per documentation there is no option that would disable this conversion happening only for code blocks. Please fix this.

**Comment by tivie** [comment]
That is the expected behavior.

`<`, `>` and `&` need to be escaped in HTML, `<code>` tags are not an exception.

> You always represent an ampersand in HTML content as `&amp;` or `&#38;`, never a bare &. This includes both text between an element's tags, as well as attribute values. No exceptions.
>This is because & itself denotes the start of an HTML entity reference (as seen above) and can often cause problems when appearing in the middle of text that could be interpreted as an entity reference.

**Comment by scssyworks** [comment]
Hello Tivie,

I want to display HTML text within code block. Example Markdown:

\<code>
 \<div class="someClass">This is some HTML\<\div>
\</code>

Now as per functionality I think it is fine. I can simply write the HTML text as above (without escaping <,>, and &, and it will render correctly. However, if you open the same markdown on Github, it interprets it as proper HTML and it will display the text as follows:

This is some HTML

No wrapping div displayed.

I will have to escape it separately on github to make it work. For example, same code above need to be written as follows to make it work:

\<code>
\\\<div class="someClass">This is some HTML\\\</div>
\</code>

Now showdown is interpreting it incorrectly.

Please suggest how to achieve it as there is no way to do it in showdown. Before closing the ticket kindly understand the problem. Just that it has been implemented that way doesn't mean it is correct.

**Comment by tivie** [comment]
Ah, I see. [Well, you seem to have found a related bug in the code.](http://showdownjs.github.io/demo/#/%3Cdiv%3E%0A%20%20%3Ccode%3E%0A%20%20%3Cdiv%20class%3D%22someClass%22%3EThis%20is%20some%20HTML%3C%5Cdiv%3E%0A%20%20%3C%2Fcode%3E%0A%3C%2Fdiv%3E)

**Comment by tivie** [comment]
**input:**
```md
<div>
  <code>
  <div class="someClass">This is some HTML<\div>
  </code>
</div>
```

**expected output;**

```html
<div>
  <code>
  <div class="someClass">This is some HTML<\div>
  </code>
</div>
```

**actual output**:

```
<div>
  <code><p></p>
  </code><div class="someClass"><code>This is some HTML&lt;\div&gt;
  </code>
</div></div>
```

**Comment by tivie** [comment]
Actually, there is no bug.

If you want a span element to be parsed as is, you can wrap it in a block (a div for instance)

```
<div>
  <code>
  <div class="someClass">This is some HTML</div>
  </code>
</div>
```

**Comment by scssyworks** [comment]
This seem to be a workaround for the problem. Anyways, thanks.

**Comment by tivie** [comment]
Well, it's not a problem "per se".

There are two kinds of HTML elements in showdown, semantically speaking):

 - **span** (inline html elements)
 - **block**

Block elements are not parsed unless you add `markdown=1` attribute to it.
Span elements are always parsed, unless they are inside a block

This happens for a number of reasons:

 - **historical**: the original parser, written by John Gruber, worked like that so showdown replicated it.

 - **pratical**: almost [all parsers behave like that](https://johnmacfarlane.net/babelmark2/?normalize=1&text=%3Cspan%3E**bazinga**%3C%2Fspan%3E%0A%0A%3Cdiv%3E**bazinga**%3C%2Fdiv%3E), so people expect this kind of behavior

- **tehcnical**: since span elements can appear interpolated in text, it would be hard to parse them

**Comment by 3stacks** [comment]
Rather than creating a new one, I might piggy-back off this issue because it's sort of the same issue.

Why is it un-reasonable to expect...
```
```
<div>
    <p> This code snippet </p>
</div>
```
```
To be rendered as <code> elements with HTML inside it?

I need a library that can handle this as all of my markdown content is related to code, so there's a lot of css, javascript and html floating around. 

If a Markdown parsing library can't handle this, then I legitimately don't know what I can do.

**Comment by tivie** [comment]
Because almost all use cases expect code inside `<pre><code>` tags to be a respresentation of said code.

For instance, if the code is not escaped...

this:
```
<pre><code>
<div>foobar</div>
</code></pre>
```

is interpreted as this:

<pre><code>
<div>foobar</div>
</code></pre>
